diff/diff_test.go

```diff
@@ -144,11 +146,13 @@ func TestDiffListOrder(t *testing.T) {
 		map[string]interface{}{"__key": "1"},
 		map[string]interface{}{"__key": "2"},
 		map[string]interface{}{"__key": "3"},
+		nil,
```

Comment:
have you thought about whether nil should be here or at the beginning? should we think about whether `merge` reverses this correctly? should we worry about client-side implementation of merge as well?

Reply:
If I put the `nil` at the beginning the test fails. I didn't consider this at all. Is there a way we can prevent `nil` from making it in here?

Reply:
@changpingc this has been addressed.